units/unit.cpp

```diff
@@ -3,79 +3,47 @@
 /// @file
 /// @author Simon Heybrock
 /// @author Neil Vaytet
+#include <regex>
 #include <stdexcept>
-#include <vector>
 
-#include <boost/lexical_cast.hpp>
-#include <boost/units/cmath.hpp>
-#include <boost/units/io.hpp>
+#include <units/units.hpp>
 
-#include "scipp/units/boost_units_util.h"
 #include "scipp/units/except.h"
 #include "scipp/units/unit.h"
 
 namespace scipp::units {
 
-// Helper to check whether type is a member of a given std::tuple
-template <typename T, typename VARIANT_T> struct isTupleMember;
-template <typename T, typename... ALL_T>
-struct isTupleMember<T, std::tuple<ALL_T...>>
-    : public std::disjunction<std::is_same<T, ALL_T>...> {};
-// Helper to make checking for allowed units more compact
-template <class Units, class T> constexpr bool isKnownUnit(const T &) {
-  return isTupleMember<T, Units>::value;
-}
-
-template <class... Ts> auto make_name_lut(std::tuple<Ts...>) {
-  return std::array{std::string(boost::lexical_cast<std::string>(Ts{}))...};
+Unit::Unit(const std::string &unit)
+    : Unit(llnl::units::unit_from_string(unit == "dimensionless" ? "" : unit)) {
 }
 
 std::string Unit::name() const {
-  static const auto names = make_name_lut(supported_units_t{});
-  return names[index()];
+  auto repr = to_string(m_unit);
+  // Replace 'LATIN CAPITAL LETTER A WITH RING ABOVE' by 'ANGSTROM SIGN'
+  repr = std::regex_replace(repr, std::regex("Å"), "Å");
+  repr = std::regex_replace(repr, std::regex("^u"), "µ");
+  repr = std::regex_replace(repr, std::regex("item"), "counts");
+  return repr == "" ? "dimensionless" : repr;
 }
```

Comment:
do we need to worry about the inverse operation of this i.e. creating Unit from name str. If so, maybe best to define a common map (or vector of pairs) that can be used both ways?

Reply:
Reverse seems to work. But my goal is to make things configureable upstream, so string reprs of certain units can be defined by the lib user. For example, they use `hr` for formatting hours, but I would like to have `h` (SI).